skfp/preprocessing/filters/tice_herbicides.py

```diff
@@ -0,0 +1,96 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumHBA, CalcNumHBD, CalcNumRotatableBonds
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class TiceHerbicidesFilter(BaseFilter):
+    r"""
+    Tice rule for herbicides.
+
+    Rule established based on statistical analysis of herbicide molecules [1]_.
+
+    Molecule must fulfill conditions:
+
+        - 150 <= molecular weight <= 500
+        - logP <= 3.5
+        - HBD <= 3
+        - 2 <= HBA <= 12
+        -  number of rotatable bonds <= 11
+
+    Parameters
+    ----------
+    allow_one_violation : bool, default=True
+        Whether to allow violating one of the rules for a molecule. This makes the
+        filter less restrictive, and is the part of the original definition of this
+        filter.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel. :meth:`transform_x_y` and
+        :meth:`transform` are parallelized over the input molecules. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors. See Scikit-learn documentation on ``n_jobs`` for more details.
+
+    batch_size : int, default=None
+        Number of inputs processed in each batch. ``None`` divides input data into
+        equal-sized parts, as many as ``n_jobs``.
+
+    verbose : int, default=0
+        Controls the verbosity when generating conformers.
+
+    References
+    -----------
+    .. [1] `Tice, C.M.,
+        "Selecting the right compounds for screening:
+        does Lipinski's Rule of 5 for pharmaceuticals apply to agrochemicals?."
+        Pest. Manag. Sci., 57: 3-16.
+        <https://doi.org/10.1002/1526-4998(200101)57:1\<3::AID-PS269\>3.0.CO;2-6>`_
```

Comment:
Error: DOI not found

Reply:
this url is working

Reply:
Proper one is https://doi.org/10.1002/1526-4998(200101)57:1<3::AID-PS269>3.0.CO;2-6, without one "\" in the middle. Its should work with r-string. Or does it not, and we need this \ for escaping here?

Reply:
from documentation level it works properly